tripal_chado/tests/src/Kernel/Plugin/ChadoBuddy/ChadoPropertyBuddyTest.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Drupal\Tests\tripal_chado\Kernel\Plugin\ChadoBuddy;
+
+use Drupal\Tests\tripal_chado\Kernel\ChadoTestKernelBase;
+use Drupal\tripal_chado\ChadoBuddy\Exceptions\ChadoBuddyException;
+use Drupal\tripal_chado\Database\ChadoConnection;
+
+/**
+ * Tests the Chado Property Buddy.
+ *
+ * @group ChadoBuddy
+ */
+class ChadoPropertyBuddyTest extends ChadoTestKernelBase {
+  protected $defaultTheme = 'stark';
+
+  protected ChadoConnection $connection;
+
+  protected static $modules = ['system', 'user', 'file', 'tripal', 'tripal_chado'];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Open connection to a test Chado
+    $this->connection = $this->getTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);
+
+    // Create some simple project records
+    $query = $this->connection->insert('1:project')
+      ->fields(['name' => 'proj001'])
+      ->execute();
+      $query = $this->connection->insert('1:project')
+      ->fields(['name' => 'proj002'])
+      ->execute();
+  }
+
+  /**
+   * Tests the getProperty(), insertProperty(), updateProperty(), deleteProperty() methods.
+   */
+  public function testPropertyMethods() {
+
+    $type = \Drupal::service('tripal_chado.chado_buddy');
+    $instance = $type->createInstance('chado_property_buddy', []);
+
+    // TEST: if there is no record then it should return false when we try to get it.
+    $chado_buddy_records = $instance->getProperty('project', 1, ['projectprop.type_id' => 1]);
+    $this->assertIsArray($chado_buddy_records, 'We did not retrieve an array for a property that does not exist');
+    $this->assertEquals(0, count($chado_buddy_records), 'We did not retrieve an empty array for a property that does not exist');
+
+    // TEST: We should be able to insert a property record if it doesn't exist.
+    $chado_buddy_record = $instance->insertProperty('project', 1, ['projectprop.type_id' => 1, 'projectprop.value' => 'prop001'], []);
+    $this->assertIsObject($chado_buddy_record, 'We did not insert a new property "prop001"');
+    $values = $chado_buddy_record->getValues();
+    $base_table = $chado_buddy_record->getBaseTable();
+    $schema_name = $chado_buddy_record->getSchemaName();
+    $this->assertIsArray($values, 'We did not retrieve an array of values for the new property "prop001"');
+    $this->assertEquals(28, count($values), 'The values array is of unexpected size for the new property "prop001"');
+    $pkey_id = $chado_buddy_record->getValue('projectprop.projectprop_id');
+    $this->assertTrue(is_numeric($pkey_id), 'We did not retrieve an integer pkey_id for the new property "prop001"');
+    $this->assertEquals('project', $base_table, 'The base table is incorrect for the new property "prop001"');
+    $this->assertTrue(str_contains($schema_name, '_test_chado_'), 'The schema is incorrect for the new property "prop001"');
```

Comment:
The schema name for the test instance is always stored in a property. Thus this test would be better as:
```suggestion
    $this->assertEquals($this->testSchemaName, $schema_name, 'The schema is incorrect for the new property "prop001"');
```

Reply:
Not committing this because I move this assertion into a function, but it is incorporated